storage/src/mmr/mem.rs

```diff
@@ -168,15 +168,16 @@ impl<H: CHasher> Mmr<H> {
         element_pos
     }
 
-    /// Pop the most recent leaf element out of the MMR (if it exists).
+    /// Pop the most recent leaf element out of the MMR if it exists, returning Empty or
+    /// ElementPruned errors otherwise.
     pub fn pop(&mut self) -> Result<u64, Error> {
         if self.size() == 0 {
             return Err(Empty);
```

Comment:
nit: I've more commonly seen `Result<Option<u64>, Error>` if "Empty" is considered ok.